dev-utils/run-script.js

```diff
@@ -24,42 +24,38 @@
  */
 
 const { join } = require('path')
+const runAll = require('npm-run-all')
 const {
   runKarma,
   runSauceConnect,
   runE2eTests: runE2eTestsUtils,
   buildE2eBundles: buildE2eBundlesUtils
 } = require('./test-utils')
-const runAll = require('npm-run-all')
-
 const { startTestServers } = require('./test-servers')
-
-const {
-  getSauceConnectOptions,
-  getTestEnvironmentVariables
-} = require('./test-config')
+const { getTestEnvironmentVariables } = require('./test-config')
 const { generateNotice } = require('./dep-info')
 
 const PROJECT_DIR = join(__dirname, '../')
-
-const sauceConnectOpts = getSauceConnectOptions()
 const { sauceLabs } = getTestEnvironmentVariables()
 
-function runUnitTests(launchSauceConnect = 'false', directory) {
-  const karmaConfigFile = join(__dirname, '..', directory, './karma.conf.js')
-  if (launchSauceConnect === 'true' && sauceLabs) {
-    return runSauceConnect(sauceConnectOpts, () => runKarma(karmaConfigFile))
+function runUnitTests(packagePath, startSauceConnect = 'false') {
+  const karmaConfigFile = join(PROJECT_DIR, packagePath, 'karma.conf.js')
+  if (startSauceConnect === 'true') {
+    return launchSauceConnect(() => runKarma(karmaConfigFile))
   }
   runKarma(karmaConfigFile)
 }
 
-function launchSauceConnect() {
+/**
+ * Checks if the MODE is set to "saucelabs" and decides where to run the
+ * corresponding callback function
+ */
+function launchSauceConnect(callback = () => {}) {
   if (sauceLabs) {
-    return runSauceConnect(sauceConnectOpts, () => {
-      console.log('Launched SauceConnect!')
-    })
+    return runSauceConnect(callback)
   }
-  console.log('set MODE=saucelabs to launch sauce connect')
+  console.info('Skipping sauce tests, MODE is not set to saucelabs')
```

Comment:
As far as I can see `launchSauceConnect` doesn't deal with running tests, isn't that correct?

Maybe the log should be moved to `runSauceTests`

Reply:
launchSauceConnect determines if the test should be run on saucelabs or not, so ideally the log should be present here.

Reply:
I'm not sure I follow, how exactly `launchSauceConnect` decides the tests should be run on saucelabs or not?

Reply:
checking the `saucelabs` flag using mode. https://github.com/elastic/apm-agent-rum-js/blob/1ee93d8cbb4b757e1f432de7f12c5ea4b0db1c98/dev-utils/run-script.js#L54

Reply:
But we also have this one which checks for `sauceLabs`:
https://github.com/elastic/apm-agent-rum-js/blob/1ee93d8cbb4b757e1f432de7f12c5ea4b0db1c98/dev-utils/run-script.js#L78

Reply:
Ah that one is for deciding between to run Unit tests on local or unit, e2e in saucelabs. 

The reason why I had a inner condition is because we cannot determine what to run depending on local/ci environment without checking the mode. We can do two things

1. The change present right now
2. Check for saucelabs flag on each function (`runUnitTests` and `runSauceTests`) and call launchSauceConnect accordingly

WDYT?

Reply:
I think option 2. makes sense, but let's discuss it separately.